src/scheduler.cpp

```diff
@@ -0,0 +1,143 @@
+#ifndef TEST_VERSION
+#include "Particle.h"
+#else
+#include "../tests/scheduler_test_system.hpp"
```

Comment:
Do not use relative includes - use the `include_directories` directive in cmakelists